plugins/mac-capture/mac-sck-video-capture.m

```diff
@@ -142,8 +142,14 @@ static bool init_screen_stream(struct screen_capture *sc)
                     }
                 }
             } else {
-                target_window = [sc->shareable_content.windows objectAtIndex:0];
-                sc->window = target_window.windowID;
+                target_window = nil;
```

Comment:
`target_window` should not be anything _but_ `nil` at this point (the condition for the only code changing the pointer value has not been met), so I don't think setting it to `nil` again is necessary.

Reply:
The observed _root cause_ of the bug is correct, but this change also doesn't fix the issue:

When an invalid non-zero window ID is retained in `sc->window`, `target_window` would still be `nil`. The same would be true if `sc->window` contains an actual `0`.

So the easiest fix would be to change the `else` branch into an explicit check `target_window == nil` and then fallback to the first window ID found in `sc->shareable_content.windows` (functionality which this PR removes).